examples/patchwork_example.cc

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <albatross/Tune>
+
+#include <albatross/PatchworkGP>
+
+#include <csv.h>
+#include <fstream>
+#include <gflags/gflags.h>
+#include <iostream>
+
+#define EXAMPLE_SLOPE_VALUE 0.
+#define EXAMPLE_CONSTANT_VALUE 0.
+
+#include "sinc_example_utils.h"
+
+DEFINE_string(input, "", "path to csv containing input data.");
+DEFINE_string(output, "", "path where predictions will be written in csv.");
+DEFINE_int32(n, 10, "number of training points to use.");
+DEFINE_int32(k, 5, "number of training points to use.");
+DEFINE_bool(tune, false, "a flag indication parameters should be tuned first.");
+
+using albatross::get_tuner;
+using albatross::ParameterStore;
+using albatross::RegressionDataset;
+
+struct PatchworkFunctions {
+
+  double width = 2.;
+
+  long int grouper(const double &x) const {
+    return lround(x / width);
+  }
+
+  std::vector<double> boundary(long int x, long int y) const {
+    if (fabs(x - y) == 1) {
+      double center = width * 0.5 * (x + y);
```

Comment:
yeah that's actually what we expect, since the `grouper` method rounds the intervals will be `[1, 3], [3, 5], ....` so in your example the boundary we're looking for is `3`.  I guess the word `center` might be confusing, I set it as the center of the boundary.  But then I placed more constraints on either side of that to enforce smoothness as well as continuity.